app/controllers/admin/petitions_controller.rb

```diff
@@ -61,14 +61,27 @@ def assign_email_signees_param
     return unless params[:petition]
     params[:petition][:email_signees] = params[:petition][:email_signees] == '1'
   end
-  
+
   def params_for_update_response
     assign_email_signees_param
     params.
       require(:petition).
       permit(:internal_response, :response_required, :response, :response_summary, :email_signees)
   end
-  
+
+  def edit_scheduled_debate_date
+    @petition = Petition.find(params[:id])
+  end
+
+  def update_scheduled_debate_date
+    @petition = Petition.find(params[:id])
+    if @petition.update(update_scheduled_debate_date_params)
+      redirect_to edit_response_admin_petition_path(@petition), notice: "Scheduled debate date was successfully updated."
+    else
+      render :edit_scheduled_debate_date
+    end
+  end
```

Comment:
Not worth changing I don't think, but as an FYI: I'd have done these on a new controller to keep the existing controller smaller.